File: internal/domain/ocr.go
package domain

// OCRResult contains structured data extracted from a document scan.
type OCRResult struct {
	VendorName     string
	VendorICO      string
	VendorDIC      string
	InvoiceNumber  string
	IssueDate      string // YYYY-MM-DD
	DueDate        string // YYYY-MM-DD
	TotalAmount    Amount // in halere
	VATAmount      Amount // in halere
	VATRatePercent int
	CurrencyCode   string
	Description    string
	Items          []OCRItem
	RawText        string
	Confidence     float64 // 0.0-1.0
}

// OCRItem represents a single line item extracted from a document.
type OCRItem struct {
	Description    string
	Quantity       Amount // in cents (e.g. 100 = 1.00)
	UnitPrice      Amount // in halere
	VATRatePercent int
	TotalAmount    Amount // in halere
}

File: rust/crates/zfaktury-domain/src/ocr.rs
use crate::amount::Amount;

/// Structured data extracted from a document scan.
#[derive(Debug, Clone)]
pub struct OCRResult {
    pub vendor_name: String,
    pub vendor_ico: String,
    pub vendor_dic: String,
    pub invoice_number: String,
    /// YYYY-MM-DD format.
    pub issue_date: String,
    /// YYYY-MM-DD format.
    pub due_date: String,
    pub total_amount: Amount,
    pub vat_amount: Amount,
    pub vat_rate_percent: i32,
    pub currency_code: String,
    pub description: String,
    pub items: Vec<OCRItem>,
    pub raw_text: String,
    pub confidence: f64,
}

/// A single line item extracted from a document.
#[derive(Debug, Clone)]
pub struct OCRItem {
    pub description: String,
    pub quantity: Amount,
    pub unit_price: Amount,
    pub vat_rate_percent: i32,
    pub total_amount: Amount,
}
